src/domain/order-book/stock-manager.ts

```diff
@@ -35,20 +35,21 @@ export class StockManager {
 
       // INFO: keep in mind that the allocation amount is a percentage
       // splitStock is the amount of stock required to fulfill allocation
-      const splitStock = Math.ceil(
+      const splitStock = Math.round(
```

Comment:
we might want to check with business if we get the upper bound or the lower bound when getting percentage of available carbon units.